[FIX] default_warehouse_from_sale_team: xpath dependency on flaky element

carralc opened the pull request:
This commit fixes an xpath expression that relies on a `page` element that can later be removed by the enterprise module `website_crm_score`. When trying to add the previous module as a dependency for another module, it would prevent this module from loading with an "xpath not found" error.

luisg123v reviewed (changes requested):
title: REF ->  FIX, as you're not refactoring the module but fixing an issue

BTW, inside is not exactly the same as after, as inside will put the page at the end, while after would put it just after members. In this case is OK because there aren't other modules adding pages, but it's a good practice to support that.

The proper fix would be:
- Make a PR to enterprise to fix that replace, to make it invisible instead of replaced (ideal solution but time consuming)
- Decrease priority of this view below 14 (which is the default one), so this one is applied before the enterprise one

carralc committed:
[FIX] default_warehouse_from_sale_team: xpath dependency on flaky element

This commit fixes an xpath expression that relies on a page element
that can later be removed by the enterprise module website_crm_score.
When trying to add the previous module as a dependency for another module,
it would prevent this module from loading with an "xpath not found" error.
Changint the priority of this view makes it evaluate before the
`enterprise` one and renders correctly.

carralc commented:
> title: REF -> FIX, as you're not refactoring the module but fixing an issue
> 
> BTW, inside is not exactly the same as after, as inside will put the page at the end, while after would put it just after members. In this case is OK because there aren't other modules adding pages, but it's a good practice to support that.
> 
> The proper fix would be:
> 
> * Make a PR to enterprise to fix that replace, to make it invisible instead of replaced (ideal solution but time consuming)
> * Decrease priority of this view below 14 (which is the default one), so this one is applied before the enterprise one

I still think changing the xpath expression is the correct way to go. If you have a clean instance with `website_crm_score` installed and you try to install `default_warehouse_from_sale_team`, installation will fail with the same missing xpath error. This fix also makes the view independent on further changes in both odoo and enterprise codebases.

luisg123v commented:
> If you have a clean instance with `website_crm_score` installed and you try to install `default_warehouse_from_sale_team`, installation will fail with the same missing xpath error.

Are you sure? Since the priority is less than the  default one, this view should be applied before that one.

But even if it fails, it's not this module's fault, is the enterprise one that shouldn't replace elements, specially with a so low priority.


BTW, we even have a lint for that. Replacing elements is a bad practice, it's much more preferred to hide them.

You wouldn't even need another module for that to fail. If you for instance rename the view, the old and new one will be present for a few seconds (while Odoo figures out the old one was removed), and it will also fail because the element it's trying to replace doesn't exist anymore.

luisg123v reviewed (approved):
LGTM :+1:
